test/Microsoft.TemplateEngine.Mocks/MockCreationEffects.cs

```diff
@@ -0,0 +1,83 @@
+using Microsoft.TemplateEngine.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit.Abstractions;
+
+namespace Microsoft.TemplateEngine.Mocks
+{
+    public class MockCreationEffects : ICreationEffects, ICreationEffects2, IXunitSerializable
+    {
+        public MockCreationEffects()
+        {
+
+        }
+
+        public MockCreationEffects WithPrimaryOutputs(params string[] primaryOutputs)
+        {
+            _primaryOutputs = _primaryOutputs.Concat(primaryOutputs).ToArray();
+            return this;
+        }
+
+        public MockCreationEffects WithFileChange (params MockFileChange[] fileChanges)
+        {
+            _mockFileChanges = _mockFileChanges.Concat(fileChanges).ToArray();
+            return this;
+        }
+
+        public MockCreationEffects Without (params string[] files)
+        {
+            _absentFiles = _absentFiles.Concat(files).ToArray();
+            return this;
+        }
+
+        string[] _primaryOutputs = Array.Empty<string>();
+        MockFileChange[] _mockFileChanges = Array.Empty<MockFileChange>();
+        string[] _absentFiles = Array.Empty<string>();
+
+
+        public IReadOnlyList<IFileChange> FileChanges => _mockFileChanges;
+
+        public ICreationResult CreationResult
```

Comment:
We are returning a new instance every time, which isn't `property`-like. Maybe this could be a method with a name that hints object instantiation?

Reply:
This property is part of `ICreationEffects` interface, unless it is a strong concern I would prefer to keep it as is.